nomad/structs/structs.go

```diff
@@ -360,6 +361,19 @@ type PlanRequest struct {
 	WriteRequest
 }
 
+// ApplyPlanResultsRequest is used by the planner to apply a Raft transaction
+// committing the result of a plan.
+type ApplyPlanResultsRequest struct {
+	// AllocUpdateRequest holds the allocation updates to be made by the
+	// scheduler.
+	AllocUpdateRequest
+
+	// CreatedDeployment is the deployment created as a result of a scheduling
+	// event. Any existing deployment should be cancelled when the new
+	// deployment is created.
+	CreatedDeployment *Deployment
```

Comment:
Can you update a deployment or destroy a deployment as part of a plan? Or only create?